src/control/lib/spdk/include/nvme_control.h

```diff
@@ -65,40 +68,54 @@ struct dev_health_t {
 	bool		 volatile_mem_warning;
 };
 
-/**
- * \brief NVMe controller details
- */
-struct ctrlr_t {
-	char		     model[1024];
-	char		     serial[1024];
-	char		     pci_addr[1024];
-	char		     fw_rev[1024];
-	int		     socket_id;
-	struct dev_health_t *dev_health;
-	struct ctrlr_t	    *next;
-};
-
-/**
- * \brief NVMe namespace details
- */
-struct ns_t {
-	int		id;
-	int		size;
-	char		ctrlr_pci_addr[1024];
-	struct ns_t	*next;
-};
-
 /**
  * \brief Return containing return code, controllers, namespaces and error
  * message
  */
 struct ret_t {
 	int		rc;
-	struct ctrlr_t	*ctrlrs;
-	struct ns_t	*nss;
+	struct ctrlr_t *ctrlrs;
+	struct ns_t    *nss;
 	char		err[1024];
 };
 
+struct ctrlr_entry {
+	struct spdk_nvme_ctrlr	*ctrlr;
+	struct spdk_pci_addr	 pci_addr;
+	struct dev_health_entry	*dev_health;
+	int			 socket_id;
+	struct ctrlr_entry	*next;
+};
+
+struct ns_entry {
+	struct spdk_nvme_ctrlr	*ctrlr;
+	struct spdk_nvme_ns	*ns;
+	struct ns_entry		*next;
+	struct spdk_nvme_qpair	*qpair;
+};
+
+struct dev_health_entry {
+	struct spdk_nvme_health_information_page health_page;
+	struct spdk_nvme_error_information_entry error_page[256];
+	int					 inflight;
+};
+
+extern struct ctrlr_entry	*g_controllers;
```

Comment:
nit - alignment?
Not sure I trust Github's portrayal of the tabs and spaces anymore.

Reply:
looks okay in IDE, tabwidth == 8 so I don't know, maybe gh presentation is lying